momento-sdk/src/intTest/java/momento/sdk/SimpleCacheClientTest.java

```diff
@@ -136,30 +145,22 @@ public void initializesSdkAndCanHitDataPlaneForUnreachableControlPlane() {
     try (final SimpleCacheClient client =
         SimpleCacheClient.builder(BAD_CONTROL_PLANE_JWT, DEFAULT_TTL_SECONDS).build()) {
       // Unable to hit control plane
-      final InternalServerException e =
-          assertThrows(
-              InternalServerException.class, () -> client.createCache(randomString("cacheName")));
-      assertTrue(e.getMessage().contains("Unable to reach request endpoint."));
+      final CreateCacheResponse createResponse = client.createCache(randomString("cacheName"));
+      assertThat(createResponse).isInstanceOf(CreateCacheResponse.Error.class);
+      assertThat(((CreateCacheResponse.Error) createResponse))
```

Comment:
No automatic type narrowing?

Reply:
I haven't done any error handling changes yet other than converting throws to returning an error type. i will do a separate comprehensive error handling pass.